src/cswinrt/strings/WinRT.cs

```diff
@@ -42,7 +54,74 @@ internal sealed class Platform
         internal static extern bool FreeLibrary(IntPtr moduleHandle);
 
         [DllImport("kernel32.dll", EntryPoint = "GetProcAddress", SetLastError = true, BestFitMapping = false)]
-        internal static unsafe extern void* TryGetProcAddress(IntPtr moduleHandle, [MarshalAs(UnmanagedType.LPStr)] string functionName);
+        internal static unsafe extern void* TryGetProcAddress(IntPtr moduleHandle, sbyte* functionName);
+
+        internal static unsafe void* TryGetProcAddress(IntPtr moduleHandle, ReadOnlySpan<byte> functionName)
+        {
+            fixed (byte* lpFunctionName = functionName)
+            {
+                return TryGetProcAddress(moduleHandle, (sbyte*)lpFunctionName);
+            }
+        }
+
+        internal static unsafe void* TryGetProcAddress(IntPtr moduleHandle, string functionName)
+        {
+            bool allocated = false;
+            Span<byte> buffer = stackalloc byte[0x100];
+            if (functionName.Length * 3 >= 0x100) // Maximum of 3 bytes per UTF-8 character, stack allocation limit of 256 bytes (including the null terminator)
+            {
+                // Calculate accurate byte count when the provided stack-allocated buffer is not sufficient
+                int exactByteCount = checked(Encoding.UTF8.GetByteCount(functionName) + 1); // + 1 for null terminator
+                if (exactByteCount > 0x100)
+                {
+#if NET6_0_OR_GREATER
+                    buffer = new((byte*)NativeMemory.Alloc((nuint)exactByteCount), exactByteCount);
+#else
+                    buffer = (byte*)Marshal.AllocHGlobal(exactByteCount);
+#endif
+                    allocated = true;
+                }
+            }
+
+            var rawByte = (byte*)Unsafe.AsPointer(ref MemoryMarshal.GetReference(buffer));
+
+            int byteCount;
+
+#if NETSTANDARD2_1_OR_GREATER
```

Comment:
Based on [this](https://learn.microsoft.com/en-us/dotnet/standard/frameworks#preprocessor-symbols), I am not sure if `NETSTANDARD2_1_OR_GREATER` will be set or not when targeting .NET 6.  To differentiate between our .NET Standard projection and our .NET5+ projection, we typically just do `#if NET` which I think will work here.

Reply:
You're right. Fixed now.